### PR TITLE
support passing most qs options to restify.queryParser

### LINKED ISSUE #925: allow passing of qs options
Problem:

qs limits the number of array elements in a URL query to 20 - above that, it will be converted to object properties, see: https://github.com/hapijs/qs.
Highly undesirable for some cases.

> qs will also limit specifying indices in an array to a maximum index of 20. Any array members with an index of greater than 20 will instead be converted to an object with the index as the key.
> This limit can be overridden by passing an arrayLimit option:

if the options which can be passed here:

server.use(restify.queryParser({...}));

would allow for a qs section or some argument pass-through mechanism, qs could be configured by the restify user. Even a specific: arrayLimit option would work.

If this request is found useful, absolutely willing to make a pull request - not sure of the standard process here. It's such a minor thing that the whole pull request/review cycle may be a bit overkill?

(note: I am aware that it's not hard to replace the restify queryParser - so it's not that there is no solution. But would consider configuration options for the built-in parser the superior solution)

### PR DESCRIPTION
**Pull request by trentm**
Fixes #925

### CODE REVIEW

**Comment by trentm** [comment]
@yunong @DonutEspresso poke. Thanks.

My intention is to:
- get this in and publish restify@4.2.0
- submit a PR to restify-plugins to allow passing the same options to `queryParser` there.

Yunong, if you recall, you and I discussed this work ages ago in private chat.

**Comment by trentm** [comment]
Oh, also, I'd like to do the 'make cutarelease' Makefile task for node-restify.git as I did for restify/clients.git (https://github.com/restify/clients/commit/d691925d10bda2ad885b6de754be26af7975da5b)

File: lib/plugins/query.js [hunk 1]
```diff
@@ -22,14 +39,33 @@ function queryParser(options) {
...
+     * compatibility for the restify 4.x stream while using the latest qs
+     * version, we need to maintain those defaults. Note that restify-plugins
+     * changes back to the pre-restify-4.x behaviour. See test/query.test.js
+     * for more details.
+     */
+    var qsOptions = {
+        plainObjects: true,
+        allowDots: true
+    };
+    Object.keys(EXPOSED_QS_OPTIONS).forEach(function (k) {
```
Comment (yunong) [inline lib/plugins/query.js]: Not sure about the performance impact of Object.keys, but presumably you could use https://www.npmjs.com/package/lodash.foreach to optimize this.

File: lib/plugins/query.js [hunk 2]
```diff
@@ -22,14 +39,33 @@ function queryParser(options) {
...
+        EXPOSED_QS_OPTIONS[k](options[k], k);
```
Comment (yunong) [inline lib/plugins/query.js]: Mind dropping a comment to describe what's happening here?

Comment (trentm) [inline lib/plugins/query.js]: done

**Comment by yunong** [comment]
LGTM, feel free to merge.

**Comment by trentm** [comment]
Thanks!

**Comment by trentm** [comment]
restify@4.2.0 published with this

**Pull request merged**